Podfile

```diff
@@ -63,6 +63,7 @@ post_install do |installer_representation|
       # let Xcode decide what archs are built
       # this is an Xcode settings recommendation
       config.build_settings.delete('ARCHS')
+      config.build_settings['MACOSX_DEPLOYMENT_TARGET'] = 10.12
```

Comment:
`platform` specifies the [platform for which a static library should be built](https://github.com/CocoaPods/Core/blob/61d22428211654717845806b701806fa1764f05f/lib/cocoapods-core/podfile/dsl.rb#L533), but not all the [targets](https://github.com/CocoaPods/Core/blob/71a32301ade4aeba13e9eaaf20235e9039fc6aa5/lib/cocoapods-core/podfile/target_definition.rb#L649). The `MACOSX_DEPLOYMENT_TARGET` is set by the `podspec` and will overwrite any changes every time you `pod install`.

This is why I like to commit the `Pods.xcodeproj` to git :)

Options are the current setting, or:

`config.build_settings.delete('MACOSX_DEPLOYMENT_TARGET')`

Then the targets are all set to the value in `platform`.

What's the issue with committing `Pods.xcodeproj` to git?

Reply:
Mainly hundreds of merge conflicts on that file every pod install / update... 

Okay, I installed the FMDB on my branch to see what you are trying to do... first problem is the library itself.

Can we open a PR to that library that updates the `podspec` by adding `s.osx.deployment_target = '10.11'` as their `pbxproj` file does? I can od it you don't want/have time for that. Because there are things that should be fixed on _library_ side.

In the meantime if you want, we can either use the fork we create for the PR, or we can use this solution you added, but can we `if` condition it only if the `config.build_settings['MACOSX_DEPLOYMENT_TARGET'] < 10.12` to not override anything we don't have to? :)

Reply:
I can fork, but I doubt it will be merged...

Reply:
I already have a discussion about that (https://github.com/CocoaPods/Core/pull/636)

I can fork `FMDC` under `SequelAce` and we will open a PR, if not accepted, we will stay with the fork.

Reply:
Or simply override the settings for targets below `10.12` (not all targets tho)

Reply:
`config.build_settings['MACOSX_DEPLOYMENT_TARGET'] = '10.12' if config.build_settings['MACOSX_DEPLOYMENT_TARGET'].to_f < 10.12` should do the trick :)

Reply:
> but can we `if` condition it only if the `config.build_settings['MACOSX_DEPLOYMENT_TARGET'] < 10.12`

Can't compare 10.6 vs 10.12 without converting to ints. Also, all the targets are < 10.12.....

Reply:
ohh shoot, float vs int. 😂 allrighty, merge it :)

Reply:
https://github.com/ccgus/fmdb/pull/806

Reply:
We can now point to `3e5efdf6690cf7dce0a914c370507deee8740d33` commit instead of branch and remove the post_install hook 🎉 Good job, thanks James!